### ISSUE TITLE
[1.18.2] Better Combat causes Thrashers to be nigh unkillable

### ISSUE THREAD

**Issue by Tyrus5255** [minecraftabnormals/Upgrade-Aquatic#346]
Minecraft Version: 1.18.2
Forge: 40.2.0

Mods: 
Better Combat 
Blueprint
Cloth Config
Player Animator
Upgrade Aquatic

When being grabbed by a thrasher, the player's camera gets stuck causing them to be unable to attack the thrasher. Better Combat has a config for attacking mounted entities, however despite this its basically impossible to actually hit the thrasher while grabbed.

**Comment by DTX3** [comment]
Yes I have just experienced that too. Couldn't get away and the camera was facing the opposite way. I had to use tp commands to transport away.

**Comment by son0fether** [comment]
Also true in 1.19.2.  This turns them from dangerous enemies into a one-hit kill monster, e.g. there's nothing you can do at all once grabbed.

Twilight Forest pinch beetles don't have this issue with Better Combat, and they also grab the player.  They let the camera move freely like being in a minecart, so the problem is probably just how being grabbed by a thrasher locks the camera.  I think that's intended to help the player.  A config to turn off the camera locking and use full camera rotation would probably be enough, or doing that by default.

I tried adding "upgrade_aquatic:thrasher" and "upgrade_aquatic:great_thrasher" to the Better Combat hostile_player_vehicles config list, but the behavior was the same.

I think I'm going to have to use InControl or similar to block thrasher and great thrasher spawns, which is a shame because they're cool.  I'll probably add thrasher teeth to the Drowned loot or something.

**Comment by thedeerirty** [comment]
Bumping this, I would love to see a fix on both versions. 

Using a pickaxe or something that doesn't use an animation allowed me to damage the thrasher, but they're so dangerous that surviving long enough to make a difference with this is the challenge of it.

**Comment by Aceplante** [comment]
Not an issue with this mod, theyre supposed to be difficult, also not an issue with better combat since the result is the same even if youre attacks arent slower. Just flick behind you as you attack?

**Comment by TVCJ13** [comment]
This is actually a conflict with better combat because without it you're supposed to be able to attack the thrasher freely to make it release it's grip. And by the way thrashers are incompatible with shaders too, shaders make the same bug, literally impossible to counter.

**Comment by Feyreth** [comment]
Same issue seems to still be apparent in 1.20.1, getting caught by a Thrasher immediately prevented me from attacking it. I had this happen a long time ago in 1.16 without Better Combat though.

Seeing that this is still open is there any chance we could get a config to disable the grab attack? I'd rather not have to disable Thrashers as a work around, they're a cool mob but it's annoying to die to them with no recourse.

**Comment by Krampus69** [comment]
Helo, this error has been out for 2 years now. Is it supposed to be on Upgrade Aquatic side ? Because this might be done on the Better Combat side too, perhaps, someone might want to fix this small glitch that makes thrashers invincible

**Comment by Skyler5255** [comment]
Better Combat has a config to handle hitting "ridden" entities, thrashers however supercede this and are imp

**Comment by Tyrus5255** [comment]
Better combat has contingencies for hitting enemies that your are "riding" (which more often than not means being grabbed) for those mobs it works just fine, just something with thrashers made them impossible to hit. However it's been 2 years I hope it's fixed or at least circumventable by now.

(Side note: wow GitHub has fast buttons that make my minor misclicks into permanent marks)
